### DISCUSSION TITLE
Dynamically Removing Admin Access to User with Button: Capstone and Rubber Duck

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#326] [Capstones!] [answered]
Ben Jarrett

I want my Admin to be able to assign each employee access to admin rights dynamically. 
I want this to be done with a button. 

I understand the concept of setting a Boolean of true or false for the button, but I don't exactly how to tackle the functions to grab the UID and set it as admin, other than just manually setting it up in the .ENV files of my project. 
Obviously, that is not something an admin employee will have access to, so I want to make it possible through my app. 
I did some research on this, and it seems that I might be able to do something with the Bind method and the id's of the employees, but I am not completely understanding what Bind is and does. Still trying to grasp my head around that atm. 

I am asking for actual code help and to rubber duck about this

No errors

https://www.freecodecamp.org/news/the-react-handbook-b71c27b0a795/
https://stackoverflow.com/questions/39818569/pass-id-through-on-click-react-js
https://stackoverflow.com/questions/58919550/toggle-between-admin-user-view-in-react
https://marmelab.com/react-admin/CreateEdit.html
https://github.com/marmelab/react-admin/blob/master/docs/Actions.md
https://www.digitalocean.com/community/tutorials/how-to-set-up-conditional-and-responsive-routing-with-react-router-v4

https://github.com/BenJarrett/FrontEnd-Capstone/blob/mapRefactoring/src/views/Staff.js

**Comment by drteresavasquez** [comment, marked as answer]
Decided to create user node for all users and set an admin boolean that admin can update.

**Reply by BenJarrett** [reply]
Hey Dr. T,

I have users being created with a UID that are being mapped to my "Staff" screen. 
When the user object is created, adminAccess is set to "false" .

Could we have a quick rubber duck session about the logic during class tonight? 
I tried numerous ways to make it work today with no success.
